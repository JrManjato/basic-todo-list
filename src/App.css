.App-header {
    display: flex;
}

.col {
    width: 20%;
    border: 1px solid black;
}