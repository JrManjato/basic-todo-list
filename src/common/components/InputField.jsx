import React, { useState } from 'react';

const InputField = ({ setList }) => {
    const [newItem, setNewItem] = useState('');

    const addItem = (e) => {
        e.preventDefault();
        // ! Check for empty item
        if (!newItem) {
            alert("Press enter an item.");
            return;
        }

        const item = {
            id: Date.now(),
            value: newItem,
            scale: 'todo'
        };

        // Add new item to items array
        setList((oldList) => [...oldList, item]);

        // Reset newItem back to original state
        setNewItem("");
    }
    return (
        <form
            className="form col-md-10"
            onSubmit={(e) => {
                addItem(e);
            }}
        >
            <input
                type="text"
                placeholder="Add a task..."
                value={newItem}
                onChange={(e) => setNewItem(e.target.value)}
                data-testid="input-field"
            />
        </form>
    )
}

export default InputField;