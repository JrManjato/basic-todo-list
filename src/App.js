import './App.css';
import React, { useState } from "react";
import List from './common/components/List';
import InputField from './common/components/InputField';

const App = () => {
  const [list, setList] = useState([]);

  return (
    <div className="App">
      <header className="App-header">
        <div className='col'>
          <h1>Enter a task</h1>
          <InputField setList={setList} />
        </div>
        <div className='col'>
          <h1>To do</h1>
          <List list={list} scale='todo' isCheckable={true} setList={setList} />
        </div>
        <div className='col'>
          <h1>Done</h1>
          <List list={list} scale='done' isCheckable={false} setList={setList} />
        </div>
      </header>
    </div>
  );
}

export default App;
