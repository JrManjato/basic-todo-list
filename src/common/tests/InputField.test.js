import { create } from "react-test-renderer";
import { render, fireEvent, screen } from "@testing-library/react";
import App from "../../App";
import InputField from "../components/InputField";

describe("Rendering Test", () => {

    it("should renders the component", () => {
        const renderer = create(<InputField />);

        expect(renderer).toBeTruthy();
    });

    it("should renders the component [snapshot]", () => {
        const renderer = create(<InputField />);

        const json = renderer.toJSON();

        expect(json).toMatchSnapshot();
    });
});

describe("InputField Test", () => {

    beforeEach(() => {
        render(<App />);
    });

    it("should renders the app", () => {
        const title = screen.getByText(/To do/i);

        expect(title).toBeInTheDocument();
    });

    it("should change the value on the input", () => {
        const input = screen.getByTestId("input-field");
        const inputValue = 'todo-1';

        fireEvent.change(input, { target: { value: inputValue } });

        expect(input.value).toBe(inputValue);
    });

    it("should clear the input when it is submitted", () => {
        const input = screen.getByTestId("input-field");
        const inputValue = 'todo-1';

        fireEvent.change(input, { target: { value: inputValue } });
        fireEvent.submit(input);

        expect(input.value).toBe("");
    });

    it("should add a todo when the input is submitted", () => {
        const input = screen.getByTestId("input-field");
        const inputValue = 'todo-1';

        fireEvent.change(input, { target: { value: inputValue } });
        fireEvent.submit(input);

        const listItem = screen.getByTestId("todo");

        expect(listItem.textContent).toBe(inputValue);
    });
});