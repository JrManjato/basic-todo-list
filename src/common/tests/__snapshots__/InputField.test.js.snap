// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Test should renders the component [snapshot] 1`] = `
<form
  className="form col-md-10"
  onSubmit={[Function]}
>
  <input
    data-testid="input-field"
    onChange={[Function]}
    placeholder="Add a task..."
    type="text"
    value=""
  />
</form>
`;
